frontend/src/views/Game.tsx

```diff
@@ -140,22 +173,48 @@ function GamePage() {
 
   // Send notification if submission result is correct and currentUser is set.
   const checkSendTestCorrectNotification = (submissionParam: SubmissionResult) => {
-    if (submissionParam.status === 'SUCCESS' && currentUser) {
+    if (submissionParam.numCorrect === submissionParam.numTestCases && currentUser) {
       const notificationBody: string = JSON.stringify({
         initiator: currentUser,
         time: new Date(),
         notificationType: NotificationType.TestCorrect,
-        content: submissionParam.output,
+        content: 'success',
       });
       send(routes(roomId).subscribe_notification, {}, notificationBody);
     }
   };
 
   // Callback when user runs code against custom test case
   const runSolution = (input: string) => {
-    const tempSubmission: SubmissionResult = { status: 'SUCCESS', output: input };
-    setSubmission(tempSubmission);
-    checkSendTestCorrectNotification(tempSubmission);
+    const request = {
+      initiator: currentUser!,
+      code: input,
+      language: currentLanguage,
+    };
+
+    submitSolution(roomId, request)
+      .then((res) => {
+        setSubmission(res);
+        checkSendTestCorrectNotification(res);
+      })
+      .catch((err) => setError(err));
+  };
+
+  const exitGame = () => {
+    if (window.confirm('Exit the game? You will not be able to rejoin.')) {
+      history.replace('/');
```

Comment:
Per [this discussion](https://github.com/rocketden/main/pull/93#discussion_r560563908), I think we may want to disconnect from the socket here as well.

Reply:
👍